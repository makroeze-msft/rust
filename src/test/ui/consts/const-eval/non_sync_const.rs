#![feature(optin_builtin_traits)]

struct Foo;
impl !Sync for Foo {}

struct Bar(i32);
impl !Sync for Bar {}

const FOO : &Foo = &Foo; //~ ERROR cannot borrow
const FOO2 : Option<&Foo> = Some(&Foo); //~ ERROR cannot borrow
//~^ ERROR borrowed value does not live long enough
const FOO3 : &Option<Foo> = &Some(Foo); //~ ERROR cannot borrow

const BAR : &Bar = &Bar(42); //~ ERROR cannot borrow
const BAR2 : Option<&Bar> = Some(&Bar(42)); //~ ERROR cannot borrow
//~^ ERROR borrowed value does not live long enough
const BAR3 : &Option<Bar> = &Some(Bar(42)); //~ ERROR cannot borrow

fn main() {}
