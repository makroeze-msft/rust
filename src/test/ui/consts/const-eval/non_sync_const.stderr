error[E0492]: cannot borrow a constant which may contain interior mutability or non-`Sync` data. If your data is `Sync`, create a static instead
  --> $DIR/non_sync_const.rs:9:20
   |
LL | const FOO : &Foo = &Foo; //~ ERROR cannot borrow
   |                    ^^^^

error[E0492]: cannot borrow a constant which may contain interior mutability or non-`Sync` data. If your data is `Sync`, create a static instead
  --> $DIR/non_sync_const.rs:10:34
   |
LL | const FOO2 : Option<&Foo> = Some(&Foo); //~ ERROR cannot borrow
   |                                  ^^^^

error[E0597]: borrowed value does not live long enough
  --> $DIR/non_sync_const.rs:10:35
   |
LL | const FOO2 : Option<&Foo> = Some(&Foo); //~ ERROR cannot borrow
   |                                   ^^^- temporary value only lives until here
   |                                   |
   |                                   temporary value does not live long enough
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0492]: cannot borrow a constant which may contain interior mutability or non-`Sync` data. If your data is `Sync`, create a static instead
  --> $DIR/non_sync_const.rs:12:29
   |
LL | const FOO3 : &Option<Foo> = &Some(Foo); //~ ERROR cannot borrow
   |                             ^^^^^^^^^^

error[E0492]: cannot borrow a constant which may contain interior mutability or non-`Sync` data. If your data is `Sync`, create a static instead
  --> $DIR/non_sync_const.rs:14:20
   |
LL | const BAR : &Bar = &Bar(42); //~ ERROR cannot borrow
   |                    ^^^^^^^^

error[E0492]: cannot borrow a constant which may contain interior mutability or non-`Sync` data. If your data is `Sync`, create a static instead
  --> $DIR/non_sync_const.rs:15:34
   |
LL | const BAR2 : Option<&Bar> = Some(&Bar(42)); //~ ERROR cannot borrow
   |                                  ^^^^^^^^

error[E0597]: borrowed value does not live long enough
  --> $DIR/non_sync_const.rs:15:35
   |
LL | const BAR2 : Option<&Bar> = Some(&Bar(42)); //~ ERROR cannot borrow
   |                                   ^^^^^^^- temporary value only lives until here
   |                                   |
   |                                   temporary value does not live long enough
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0492]: cannot borrow a constant which may contain interior mutability or non-`Sync` data. If your data is `Sync`, create a static instead
  --> $DIR/non_sync_const.rs:17:29
   |
LL | const BAR3 : &Option<Bar> = &Some(Bar(42)); //~ ERROR cannot borrow
   |                             ^^^^^^^^^^^^^^

error: aborting due to 8 previous errors

Some errors occurred: E0492, E0597.
For more information about an error, try `rustc --explain E0492`.
